### ISSUE TITLE
Analyzer produces lots of errors for *.html files

### ISSUE THREAD

**Issue by zoechi** [dart-lang/sdk#28905]
> Dart VM version: 1.23.0-edge.3a1b7bdfcaa9374f76807062914ebae29d389408 (Fri Feb 24 15:01:07 2017) on "linux_x64"
```html
<!DOCTYPE html>
<html>
<head>
  <title>FHIR UI Designer</title>
  <meta name="viewport" content="width=device-width, initial-scale=1">

  <script defer src="designer.dart" type="application/dart"></script>
  <script defer src="packages/browser/dart.js"></script>

  <link rel="stylesheet" href="spinner.css">
  <link rel="stylesheet" href="designer.css">
  <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons">
  <link rel="stylesheet" href="packages/fhir_designer/asset/material.cyan-light_blue.css">
  <script src="packages/bwu_ng_mdl/asset/dialog_polyfill/dialog-polyfill.js"></script>
  <link rel="stylesheet" type="text/css" href="packages/bwu_ng_mdl/asset/dialog_polyfill/dialog-polyfill.css" />

</head>
<body>
<div class="container-fluid">
  <my-app class="row">
    <div class="sk-cube-grid">
      <div class="sk-cube sk-cube1"></div>
      <div class="sk-cube sk-cube2"></div>
      <div class="sk-cube sk-cube3"></div>
      <div class="sk-cube sk-cube4"></div>
      <div class="sk-cube sk-cube5"></div>
      <div class="sk-cube sk-cube6"></div>
      <div class="sk-cube sk-cube7"></div>
      <div class="sk-cube sk-cube8"></div>
      <div class="sk-cube sk-cube9"></div>
    </div>
  </my-app>
</div>
</body>
</html>
```
```
ERROR: Expected to find ';'. ([fhir_designer] web/index.html:1)
ERROR: Unexpected text '<'. ([fhir_designer] web/index.html:1)
ERROR: Expected a method, getter, setter or operator declaration. ([fhir_designer] web/index.html:1)
ERROR: Expected a method, getter, setter or operator declaration. ([fhir_designer] web/index.html:1)
ERROR: Unexpected text '>'. ([fhir_designer] web/index.html:1)
ERROR: Undefined class 'DOCTYPE'. ([fhir_designer] web/index.html:1)
ERROR: html isn't a type. ([fhir_designer] web/index.html:2)
ERROR: Expected a method, getter, setter or operator declaration. ([fhir_designer] web/index.html:2)
ERROR: Unexpected text '>'. ([fhir_designer] web/index.html:2)
ERROR: Undefined class 'head'. ([fhir_designer] web/index.html:3)
ERROR: Expected a method, getter, setter or operator declaration. ([fhir_designer] web/index.html:3)
ERROR: Unexpected text '>'. ([fhir_designer] web/index.html:3)
ERROR: The name '=' is already defined. ([fhir_designer] web/index.html:3)
ERROR: The name '' is already defined. ([fhir_designer] web/index.html:3)
ERROR: Expected to find ';'. ([fhir_designer] web/index.html:4)
ERROR: Expected to find '>'. ([fhir_designer] web/index.html:4)
ERROR: Unexpected text '/'. ([fhir_designer] web/index.html:4)
ERROR: The name '' isn't a type so it can't be used as a type argument. ([fhir_designer] web/index.html:4)
ERROR: Expected a method, getter, setter or operator declaration. ([fhir_designer] web/index.html:4)
ERROR: Expected a method, getter, setter or operator declaration. ([fhir_designer] web/index.html:4)
ERROR: Expected a method, getter, setter or operator declaration. ([fhir_designer] web/index.html:4)
ERROR: Undefined class 'FHIR'. ([fhir_designer] web/index.html:4)
ERROR: Unexpected text '>'. ([fhir_designer] web/index.html:4)
ERROR: Undefined class 'title'. ([fhir_designer] web/index.html:4)
ERROR: Unexpected text '>'. ([fhir_designer] web/index.html:4)
ERROR: Undefined class 'title'. ([fhir_designer] web/index.html:4)
ERROR: The name '=' is already defined. ([fhir_designer] web/index.html:4)
ERROR: The name '=' is already defined. ([fhir_designer] web/index.html:4)
ERROR: The name '=' is already defined. ([fhir_designer] web/index.html:4)
ERROR: Expected a type name. ([fhir_designer] web/index.html:4)
ERROR: Undefined class 'Designer'. ([fhir_designer] web/index.html:4)
ERROR: The name '' is already defined. ([fhir_designer] web/index.html:4)
ERROR: The name '' is already defined. ([fhir_designer] web/index.html:4)
ERROR: The name '' is already defined. ([fhir_designer] web/index.html:4)
ERROR: Undefined class 'meta'. ([fhir_designer] web/index.html:5)
ERROR: The operator '>' isn't defined for the class 'String'. ([fhir_designer] web/index.html:5)
ERROR: Expected to find ';'. ([fhir_designer] web/index.html:5)
ERROR: Missing variable type for 'content'. ([fhir_designer] web/index.html:5)
ERROR: Variables must be declared using the keywords 'const', 'final', 'var' or a type name. ([fhir_designer] web/index.html:5)
ERROR: Expected to find '>'. ([fhir_designer] web/index.html:7)
ERROR: The operator '>' isn't defined for the class 'String'. ([fhir_designer] web/index.html:7)
ERROR: Expected to find ';'. ([fhir_designer] web/index.html:7)
ERROR: Expected to find ';'. ([fhir_designer] web/index.html:7)
...
```

**Comment by bwilkerson** [comment]
I saw the same thing this weekend.

**Comment by scheglov** [comment]
https://codereview.chromium.org/2720023002

**Comment by zoechi** [comment]
The errors are gone. I guess this can be closed.

**Comment by bwilkerson** [comment]
Thanks for confirming!

**Comment by alexander-doroshko** [comment]
@scheglov I see this bug in SDK 1.23-dev.9.0 that was released much later than this fix landed. But no problems with today's SDK 1.23-edge build. Just wanted to make sure the fix will appear in the next SDK dev release.

**Comment by alexander-doroshko** [comment]
@scheglov, sorry for bothering you. The fix doesn't seem to be included in the 1.23-dev.9.1 from http://gsdview.appspot.com/dart-archive/channels/dev/raw/1.23.0-dev.9.1/sdk/
![image](https://cloud.githubusercontent.com/assets/4499789/24196584/55c5fd76-0f0f-11e7-8c59-568cdacb6c53.png)

**Comment by scheglov** [comment]
Yes, I think you're correct - [the fix](https://github.com/dart-lang/sdk/commit/319b726f8b18451bd02645e1cc3bf8e5713780f2) has not been released yet. There are no other tags than "master".
![image](https://cloud.githubusercontent.com/assets/384794/24213795/ef54c586-0ef0-11e7-8267-e1bd88378603.png)

**Comment by alexander-doroshko** [comment]
Thanks for the correct link to the fix! Then I ping @MichaelRFairhurst, please make sure the fix lands in 1.23-dev builds. :)
